tests/app/celery/test_scheduled_tasks.py

```diff
@@ -397,3 +403,142 @@ def test_check_templated_letter_state_during_utc(mocker, sample_letter_template)
         subject="[test] Letters still in 'created' status",
         ticket_type='incident'
     )
+
+
+# Setup a pytest fixture to mock the DynamoDB table
+@pytest.fixture
+def dynamodb_mock():
+    with mock_dynamodb():
+        dynamodb = boto3.resource('dynamodb', region_name='us-east-1')
+
+        # Create a mock DynamoDB table
+        table = dynamodb.create_table(
+            TableName='TestTable',
+            KeySchema=[
+                {
+                    'AttributeName': 'id',
+                    'KeyType': 'HASH'
+                }
+            ],
+            AttributeDefinitions=[
+                {
+                    'AttributeName': 'id',
+                    'AttributeType': 'S'
+                }
+            ],
+            ProvisionedThroughput={
+                'ReadCapacityUnits': 1,
+                'WriteCapacityUnits': 1
+            }
+        )
+
+        # Wait for table to be created
+        table.meta.client.get_waiter('table_exists').wait(TableName='TestTable')
+
+        yield table
+
+
+def test_get_dynamodb_comp_pen_messages_with_empty_table(dynamodb_mock):
+    # Invoke the function with the mocked table and application
+    messages = _get_dynamodb_comp_pen_messages(dynamodb_mock, message_limit=1)
+
+    assert messages == [], "Expected no messages from an empty table"
+
+
+def test_get_dynamodb_comp_pen_messages_with_data(dynamodb_mock):
+    message_limit = 3
+    # Insert mock data into the DynamoDB table
+    items_to_insert = [
+        {'id': '1', 'is_processed': False},
+        {'id': '2', 'is_processed': False},
+        {'id': '3', 'is_processed': False},
+        {'id': '4', 'is_processed': False},
+        {'id': '5', 'is_processed': False},
+    ]
+
+    with dynamodb_mock.batch_writer() as batch:
+        for item in items_to_insert:
+            batch.put_item(Item=item)
```

Comment:
Please make the insertion a fixture generator e.g.
```
def sample_dynamodb_insert(dynamo_mock):
    inserted_items = []
    def wrapper(items_to_insert: dict)
        do the insert
        add to inserted list
    
    yield wrapper
    
    # Teardown
    Clean out the inserted entries
```

This will prevent the items from persisting in the table outside the intended function, thus reducing the chance of side-effects/bugs.

Reply:
added new fixture for this